xcodeproj/internal/bazel_integration_files/bazel_build.sh

```diff
@@ -0,0 +1,204 @@
+    readonly output_group_prefixes="bc"
+  fi
+else
+  if [[ "$RULES_XCODEPROJ_BUILD_MODE" == "xcode" ]]; then
+    # Inputs for compiling, inputs for linking
+    readonly output_group_prefixes="xc,xl"
+  else
+    # Compiled outputs (i.e. swiftmodules) and products (i.e. bundles)
+    readonly output_group_prefixes="bc,bp"
+  fi
+fi
+
+# We need to read from `$output_groups_file` as soon as possible, as concurrent
+# writes to it can happen during indexing, which breaks the off-by-one-by-design
+# nature of it
+IFS=$'\n' read -r -d '' -a output_groups < \
+  <( "$CALCULATE_OUTPUT_GROUPS_SCRIPT" \
+       "$ACTION" \
+       "$non_preview_objroot" \
+       "$base_objroot" \
+       "$scheme_target_ids_file" \
+       $output_group_prefixes \
+       && printf '\0' )
+
+if [ -z "${output_groups:-}" ]; then
+  if [ "$ACTION" == "indexbuild" ]; then
+    if [[ "$RULES_XCODEPROJ_BUILD_MODE" == "xcode" ]]; then
+      output_groups=("all_xc")
+    else
+      echo "error: Can't yet determine Index Build output group." \
+"Next build should succeed. If not, please file a bug report here:" \
+"https://github.com/buildbuddy-io/rules_xcodeproj/issues/new?template=bug.md." \
+        >&2
+      exit 1
+    fi
+  else
+    echo "error: BazelDependencies invoked without any output groups set." \
+"Please file a bug report here:" \
+"https://github.com/buildbuddy-io/rules_xcodeproj/issues/new?template=bug.md." \
+      >&2
+    exit 1
+  fi
+fi
+output_groups_flag="--output_groups=$(IFS=, ; echo "${output_groups[*]}")"
+
+# Set `bazel_cmd` for calling `bazel`
+
+if [ "$ACTION" == "indexbuild" ]; then
+  # We use a different output base for Index Build to prevent normal builds and
+  # indexing waiting on bazel locks from the other
+  output_base="$OBJROOT/bazel_output_base"
+fi
+
+bazelrcs=(
+  --noworkspace_rc
+  "--bazelrc=$BAZEL_INTEGRATION_DIR/xcodeproj.bazelrc"
+)
+if [[ -s ".bazelrc" ]]; then
+  bazelrcs+=("--bazelrc=.bazelrc")
+fi
+if [[ -s "$BAZEL_INTEGRATION_DIR/xcodeproj_extra_flags.bazelrc" ]]; then
+  bazelrcs+=("--bazelrc=$BAZEL_INTEGRATION_DIR/xcodeproj_extra_flags.bazelrc")
+fi
+
+bazel_cmd=(
+  env -i
+  HOME="$HOME"
+  PATH="/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
+  USER="$USER"
+  "$BAZEL_PATH"
+  "${bazelrcs[@]}"
+  ${output_base:+--output_base "$output_base"}
+)
+
+# Determine Bazel output_path
+
+if [[ "${COLOR_DIAGNOSTICS:-NO}" == "YES" ]]; then
+  color=yes
+else
+  color=no
+fi
+
+output_path=$("${bazel_cmd[@]}" \
+  info \
+  --config=rules_xcodeproj_info \
+  --color="$color" \
+  --experimental_convenience_symlinks=ignore \
+  --symlink_prefix=/ \
+  --bes_backend= \
+  --bes_results_url= \
+  output_path)
+exec_root="${output_path%/*}"
+
+# Create `bazel.lldbinit``
+
+if [[ "$ACTION" != "indexbuild" && "${ENABLE_PREVIEWS:-}" != "YES" ]]; then
+  # shellcheck disable=SC2046
+  "$BAZEL_INTEGRATION_DIR/create_lldbinit.sh" \
+    "$exec_root" \
+    $(xargs -n1 <<< "${RESOLVED_EXTERNAL_REPOSITORIES:-}") \
```

Comment:
Ideally I wouldn't have to invoke another process just to keep quoted substrings as strings, Anyone know any bash-only ways of doing this?

Reply:
What is a typical value of `RESOLVED_EXTERNAL_REPOSITORIES`? Space separated paths?